controllers/shared.go

```diff
@@ -22,7 +23,21 @@ func nodeUpdateNeedsReconcile(ev event.UpdateEvent) bool {
 	if newNode, ok = ev.ObjectNew.(*v1.Node); !ok {
 		return false
 	}
-	return conditionsNeedReconcile(oldNode.Status.Conditions, newNode.Status.Conditions)
+
+	return labelsUpdateNeedsReconcile(oldNode.Labels, newNode.Labels) ||
+		conditionsNeedReconcile(oldNode.Status.Conditions, newNode.Status.Conditions)
+}
+
+func labelsUpdateNeedsReconcile(oldLabels, newLabels map[string]string) bool {
+	// Check if the ExcludeFromRemediation label was removed
+	if _, exists := oldLabels[commonLabels.ExcludeFromRemediation]; exists {
+		// The label was present in the old labels
+		if _, exists := newLabels[commonLabels.ExcludeFromRemediation]; !exists {
```

Comment:
Nit: I think it would be a good idea to verify oldLabels/newLabels isn't nil.
I don't think it's likely to happen since that IIRC there are default labels that will always exist, but I lean onto the better safe than sorry approach here.

Reply:
not needed, because "A nil map is equivalent to an empty map except that no elements may be added." 🙂 
https://go.dev/ref/spec#Map_types

Reply:
Nice, good to know :+1: